Framework/Source/Utils/Math/ParallelReduction.cpp

```diff
@@ -136,7 +137,7 @@ namespace Falcor
         }
 
         auto texData = pRenderCtx->readTextureSubresource(mpResultFbo[mCurFbo]->getColorTexture(0).get(), 0);
-        result = *(vec4*)texData.data();
+        std::memcpy(&result[0], texData.data(), min(bytesToRead, (uint32_t)sizeof(vec4)));
```

Comment:
`memcpy` is too slow and it's performance critical code. You can get `texData` before switch and instead of calculating `bytesToRead` to the conversion to the required type